### DISCUSSION TITLE
log_figure on Azure ML

### DISCUSSION THREAD

**Discussion by RafieeAshkan** [mlflow/mlflow#7686] [Q&A] [answered]
Hi,
I am using Mlflow with pytorch Lightning and hydra for model training. All works perfectly on local machine. However, when I run on Azure ML I am facing error when overwriting images through `experiment.log_figure()`.

The error is `Exception: UserError: Resource Conflict: ArtifactId ExperimentRun/dcid.78a6d1b1-8f4f-4038-9515-f91cebb63a69/r2/validation.png already exists.`

I create and log the plot using Callback as follow:

class R2Plot(Callback):
    """Generate R2 Plots and send it to MLflow.
    Expects validation step to return predictions and targets.
    """

```
    def __init__(self):
        self.preds = []
        self.targets = []
        self.y_groups = []
        self.ready = True

    def on_validation_epoch_end(self, trainer: Trainer, pl_module: LightningModule):
        """Generate R2 plots."""
        if self.ready:
            logger = get_mlflow_logger(trainer)
            experiment = logger.experiment
            run_id = logger._run_id

            ......
            fig = pltfuncs.gen_r2(y, yhat)

            # log image as artifact
            experiment.log_figure(run_id, fig, f"r2/validation.png")

```

It generates the figure for the first epoch but for 2nd epoch, it returns the file exists error.

Would really appreciate any suggestions on how to resolve this.
Ashkan

**Comment by harupy** [comment, marked as answer]
@RafieeAshkan I think you can fix `experiment.log_figure(run_id, fig, f"r2/validation.png")` to `experiment.log_figure(run_id, fig, f"r2/validation-{epoch}.png")` to avoid `Resource Conflict`.

**Reply by RafieeAshkan** [reply]
thanks @harupy it worked by adding the epoch number to the filename. However, still would be great if I can overwrite the file just to avoid generating many files, if possible. I did not have the issue when running locally, is the issue from Azure side not allowing overwriting the file?

**Reply by harupy** [reply]
@RafieeAshkan I believe the issue is from Azure side.
